Package.swift

```diff
@@ -1,9 +1,17 @@
+// swift-tools-version:4.0
 import PackageDescription
 
 let package = Package(
     name: "MongoDriver",
+    products: [
+        .library(name: "MongoDriver", targets: ["MongoDriver"])
+    ],
     dependencies: [
-        .Package(url: "https://github.com/vapor/fluent.git", majorVersion: 2),
-        .Package(url: "https://github.com/OpenKitten/MongoKitten.git", majorVersion: 4, minor: 0),
+        .package(url: "https://github.com/vapor/fluent.git", .upToNextMajor(from: "2.4.1")),
+        .package(url: "https://github.com/OpenKitten/MongoKitten.git", .exact("4.0.16"))
```

Comment:
I'd change the MongoKitten lock to be `4.0.x` (so < `4.1.0`)

Reply:
Also, separate the Swift 4 package.swift [into a separate file](https://github.com/vapor/vapor/blob/master/Package%40swift-4.swift) so it's not a breaking change for Swift 3.1 users.

Reply:
Regarding, mongo-kitten I have tried but I am not sure how to achieve exactly that with the new swift manifest, feel free to change it as you like.

Regarding the separate package:

My understanding is that you need that only if you want to support both version at the same time.

If you leave it as it is in this pull request, people that try to install mongo-driver with swift 3.1 will be installing the latest valid tag that has a valid package manifest for 3.1, in this case 2.1.2.

Otherwise you need to add conditional defines based on the swift version, in code.